core/src/main/scala/better/files/Scanner.scala

```diff
@@ -189,6 +189,6 @@ object StringSplitter {
     * @return
     */
   def regex(pattern: String): StringSplitter = new StringSplitter {
-    override def split(s: String) = s.split(pattern, -1)
+    override def split(s: String) = s.split(pattern, -1).map(_.nn)
```

Comment:
**Java method returns array**
Java method returns array of type `Array[Null | T]`. We need to map it to `Array[T]`.